test/command/explicit-command-strategy.test.ts

```diff
@@ -0,0 +1,39 @@
+import {expect} from 'chai'
+import {resolve} from 'node:path'
+import {SinonSandbox, SinonStub, createSandbox} from 'sinon'
+import stripAnsi from 'strip-ansi'
+
+import {run, ux} from '../../src/index'
+
+describe('explicit command discovery strategy', () => {
+  let sandbox: SinonSandbox
+  let stdoutStub: SinonStub
+
+  beforeEach(() => {
+    sandbox = createSandbox()
+    stdoutStub = sandbox.stub(ux.write, 'stdout')
+  })
+
+  afterEach(() => {
+    sandbox.restore()
+  })
+
+  it('should show help for commands', async () => {
+    await run(['--help', 'foo'], resolve(__dirname, 'fixtures/explicit-commands/package.json'))
+    expect(stdoutStub.args.map((a) => stripAnsi(a[0])).join('')).to.equal(`foo topic description
+
+USAGE
+  $ oclif foo COMMAND
+
+COMMANDS
+  foo bar  foo bar description
+  foo baz  foo baz description
+
+`)
+  })
+
+  it('should run command', async () => {
```

Comment:
what happens if you typo a command path in the explicit file, so that it doesn't point to a real file?  Does `command-not-found` work there?

Reply:
Works the same as always. No command with the provided id, so this [block of code](https://github.com/oclif/core/blob/3.19.1/src/config/config.ts#L440) is run, which is what will trigger the `command_not_found` hook